src/test/java/bio/terra/service/iam/sam/SamIamTest.java

```diff
@@ -57,4 +135,274 @@ public void testExtractErrorMessageDeepNested() {
 
         assertThat(SamIam.extractErrorMessage(errorReport)).isEqualTo("FOO: BAR: (BAZ1, BAZ2: QUX)");
     }
+
+    @Test
+    public void testIgnoresNonUUIDResourceName() throws ApiException, InterruptedException {
+        final String goodId = UUID.randomUUID().toString();
+        final String badId = "badUUID";
+        when(samResourceApi.listResourcesAndPolicies(IamResourceType.SPEND_PROFILE.getSamResourceName()))
+            .thenReturn(Arrays.asList(
+                new ResourceAndAccessPolicy().resourceId(goodId),
+                new ResourceAndAccessPolicy().resourceId(badId)
+            ));
+
+        List<UUID> uuids = samIam.listAuthorizedResources(userReq, IamResourceType.SPEND_PROFILE);
+        assertThat(uuids).containsExactly(UUID.fromString(goodId));
+    }
+
+    @Test
+    public void testAuthorization() throws ApiException, InterruptedException {
+        when(samResourceApi.resourcePermissionV2(
+            IamResourceType.SPEND_PROFILE.getSamResourceName(),
+            "my-id",
+            IamAction.READ_DATA.toString()
+        )).thenReturn(true);
+        when(samResourceApi.resourcePermissionV2(
+            IamResourceType.SPEND_PROFILE.getSamResourceName(),
+            "my-id",
+            IamAction.ALTER_POLICIES.toString()
+        )).thenReturn(false);
+        assertThat(samIam.isAuthorized(userReq, IamResourceType.SPEND_PROFILE, "my-id", IamAction.READ_DATA))
+            .isTrue();
+        assertThat(samIam.isAuthorized(userReq, IamResourceType.SPEND_PROFILE, "my-id", IamAction.ALTER_POLICIES))
+            .isFalse();
+    }
+
+    @Test
+    public void testGetStatus() throws ApiException {
+        when(samStatusApi.getSystemStatus()).thenReturn(new SystemStatus()
+            .ok(true)
+            .systems(Maps.newHashMap("GooglePubSub", Maps.newHashMap("ok", true)))
```

Comment:
Another JDK 11 thing to get used to, `Map.of()`
```suggestion
            .systems(Map.of("GooglePubSub", Map.of("ok", true)))
```

Reply:
Oooh, I was looking for this. Perfect!